seqr/management/commands/check_bam_cram_paths.py

```diff
@@ -3,30 +3,47 @@
 from seqr.models import IgvSample
 from seqr.views.utils.dataset_utils import validate_alignment_dataset_path
 
+import collections
+import hail as hl
 import logging
+import tqdm
 
 logger = logging.getLogger(__name__)
 
+hl.init(log="/dev/null")
+
 
 class Command(BaseCommand):
+    help = 'Checks all gs:// bam or cram paths and, if a file no longer exists, deletes the path from the database'
 
     def add_arguments(self, parser):
-        parser.add_argument('args', nargs='*')
+        parser.add_argument(
+            '-d',
+            '--dry-run',
+            action="store_true",
+            help='Only print missing paths without updating the database',
+        )
+        parser.add_argument('args', nargs='*', help='only check paths in these project name(s)')
 
     def handle(self, *args, **options):
-        samples = IgvSample.objects.filter(
+        samples = (IgvSample.objects.filter(
             individual__family__project__name__in=args,
-        ).prefetch_related('individual', 'individual__family')
+        ) if args else IgvSample.objects.all()).prefetch_related('individual', 'individual__family__project')
 
-        failed = []
-        for sample in samples:
-            try:
-                validate_alignment_dataset_path(sample.file_path)
-            except Exception as e:
+        missing_counter = collections.defaultdict(int)
+        for sample in tqdm.tqdm(samples, unit=" samples"):
+            if sample.file_path and sample.file_path.startswith("gs://") and not hl.hadoop_is_file(sample.file_path):
```

Comment:
this should really be done at the database query level - rather than pulling everything out of the database and then checking you could do `filter(file_path__startswith='gs://')`

Reply:
oh yeah